### ISSUE TITLE
[Bug] Exported Keys of a DocumentIndex have improper format

### ISSUE THREAD

**Issue by samuelnunoo** [nextapps-de/flexsearch#260]
# Issue
Exported Keys of a DocumentIndex have improper format.
This can be reproduced by create a nested document index, exporting the keys. When you try to reimport the keys you'll see that they'll have the format "field1-field2..." instead of "field1:field2" which leads to an undefined error. 

After messing around with different key formats I found that this works
```ts

//Note result is the exported index data
    const newIndex = new Document(config)

    for (const r in result) {

     
        let string = r.replace("-",":")
        console.log(string)
        newIndex.import(string,result[r]) //
      
    }

    newIndex.search("Whales") 
```

**Comment by Himanshu21git** [comment]
You can also export your indexes in this format:
```
index.export((key, index) => {
  key = key.replaceAll('-', ':')
  localStorage.setItem(key, index)
})
```
Importing will not produce a type error now and will work the usual way.

**Comment by ts-thomas** [comment]
This is now fixed in v0.7.23
Further improvements to provide Promise.all() compatible export ist coming in next version.
